src/Features/CSharp/Portable/Completion/CompletionProviders/PartialTypeCompletionProvider.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Shared.Extensions.ContextQuery;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    internal partial class PartialTypeCompletionProvider : AbstractPartialTypeCompletionProvider
+    {
+        private static readonly SymbolDisplayFormat _symbolFormatWithGenerics =
+            new SymbolDisplayFormat(
+                globalNamespaceStyle: SymbolDisplayGlobalNamespaceStyle.Omitted,
+                typeQualificationStyle: SymbolDisplayTypeQualificationStyle.NameAndContainingTypesAndNamespaces,
```

Comment:
> As for the constraints, does VB work like C#?

Doesn't seem so. I get a compiler error unless all the declarations repeat the same set of constraints.

> Do you have tests that ensure htat we don't spit out a qualified name though from the type name itself?

We do (see `TestPartialClassWithSameMemberName`). But we also have a bug (#11569) because with the right naming, it already breaks. I don't fully understand it yet, but I think it's because the insertion position is considered _inside_ the type (another VB/C# semantic difference as far as I can tell).